src/main/java/io/confluent/kafkarest/SimpleConsumerFactory.java

```diff
@@ -59,16 +58,40 @@ private String nextClientId() {
     return id.toString();
   }
 
-  public SimpleConsumer createConsumer(final String host, final int port) {
+  public ConsumerProvider createConsumer() {
 
     final String clientId = nextClientId();
 
-    log.debug("Creating SimpleConsumer with id " + clientId + " (host: " + host + ", port: " + port + ")");
-    return new SimpleConsumer(
-        host, port,
-        simpleConsumerConfig.socketTimeoutMs(),
-        simpleConsumerConfig.socketReceiveBufferBytes(),
-        clientId);
+    log.debug("Creating SimpleConsumer with id " + clientId);
+    Properties properties = (Properties) simpleConsumerProperties.clone();
+
+    properties.setProperty("client.id", clientId);
+    properties.setProperty("max.poll.records", String.valueOf(
```

Comment:
My understanding is that Consumer object that is assigned for a single partition cannot store more records than `max.partition.fetch.bytes` (Please correct me it that is not right) so we can try to bound cache by size of records, say `max.partition.fetch.bytes` (not by number of records). In the worst case there would be number of records equal to `2 * max.partition.fetch.bytes` stored in memory for single TopicPartition (In case when the cache is full but no required offset is found in cache, consumer calls seek and poll and the second part of records is fetched.)
I will add stickiness between Cache and offset in the next commit.